src/couch_replicator_scheduler.erl

```diff
@@ -225,17 +232,22 @@ oldest_job_first(#job{} = A, #job{} = B) ->
 
 
 not_recently_crashed(#job{} = Job) ->
-    case crash_history(Job) of
-        [] ->
-            true;
-        [{crashed, When} | _] ->
+    EventsBeforeStop = lists:takewhile(
+        fun({Event, _}) -> Event =/= stopped end, Job#job.history),
+    case crash_history(EventsBeforeStop) of
+        [{{crashed, _Reason}, When} | _] = Crashes ->
+            BackoffExp = erlang:min(length(Crashes), ?MAX_BACKOFF_EXPONENT),
+            BackoffInterval = (1 bsl BackoffExp) * ?BACKOFF_INTERVAL_MICROS,
+            couch_log:notice("delay for ~p is now ~p s", [Job#job.id, BackoffInterval div 1000000]),
             timer:now_diff(os:timestamp(), When)
-                >= ?MINIMUM_CRASH_INTERVAL
+                >= BackoffInterval;
+        [] ->
+            true
     end.
 
 
 crash_history(#job{} = Job) ->
```

Comment:
I think we want `JobHistoryEvents :: list()` here as arg. Or maybe it is cleaner to push EventsBeforeStop processing to `crash_history/ 1` to make not_recently_crashed a bit smaller visually.